python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1860,6 +1863,364 @@ def delete_dynamic_partition(self, partitions_def_name: str, partition_key: str)
                 )
             )
 
+    @property
+    def supports_global_concurrency_limits(self) -> bool:
+        return self.has_table(ConcurrencySlotsTable.name)
+
+    def allocate_concurrency_slots(self, concurrency_key: str, num_int: int) -> None:
```

Comment:
this removes and adds right? so maybe `set_` / `resize_` or something instead of allocate. `num_int` -> `num_desired` 

thoughts on starting with an initial tightish limit that we can relax later instead of starting unchecked?

Reply:
yeah, +1 on the limit.